salt/_pillar/metalk8s_nodes.py

```diff
@@ -37,6 +37,30 @@ def node_info(node, ca_minion):
             if role:
                 roles.add(role)
 
+    # Introduced as a workaround in cases which bootstrap setup fails after the
+    # kubernetes API server is up and running but node role labels are still
+    # not set.
+    #
+    # The issue is that when the API server is running the ext_pillar will
+    # overwrite the hardcoded roles with the ones from the API. It will cause
+    #  bootstrap setup failure as described here:
+    # https://github.com/scality/metalk8s/issues/2137
+    #
+    # The `is_bootstrap` flag is set in pillar only when run the
+    # `metalk8s.orchestrate.bootstrap` state.
+    if not roles and pillar.get("is_bootstrap", False):
+        if "metalk8s" in pillar:
+            if "nodes" in pillar["metalk8s"]:
+                if ca_minion in pillar["metalk8s"]["nodes"]:
```

Comment:
No, why this one ? 

CA minion is handle after

Reply:
Here is the pilar we sensed to find :

```
{%- set pillar_data = {
        'bootstrap_id': pillar.bootstrap_id,
        'is_bootstrap': True,
        'metalk8s': {
            'nodes': {
                '_errors': None,
                pillar.bootstrap_id: {
                    'roles': ['bootstrap', 'master', 'etcd', 'ca', 'infra'],
                    'version': version,
                },
            },
            'volumes': None,
            'endpoints': {
                'salt-master': [{
                    'ip': bootstrap_grains['control_plane_ip'],
                    'ports': {
                        'api': 4507,
                    },
                }],
                'repositories': [{
                    'ip': bootstrap_grains['control_plane_ip'],
                    'ports': {
                        'http': 8080,
                    },
                }],
            },
        },
    }
%}
```

We need a `node name` in order to access `roles`. In this case, the node name we have is the value of `ca_minion`.

Reply:
My bad... I'll retrieve the node name from `node`...